src/main/java/nl/novi/serviceconnect/repository/ServiceRepository.java

```diff
@@ -11,5 +12,10 @@ public interface ServiceRepository extends JpaRepository<Service, Long> {
     @Query("SELECT c.name AS categoryName, COUNT(s) AS numberOfServices FROM Service s JOIN s.category c GROUP BY c.id")
     List<Object[]> countServicesPerCategory();
 
-   // @Query("SELECT * FROM Service WHERE state = \"Available\"  AND category.id =    ")
+    @Query("SELECT s FROM Service s WHERE (:categoryId IS NULL OR s.category.id = :categoryId) " +
+            "AND (:minPrice IS NULL OR s.price >= :minPrice) " +
+            "AND (:maxPrice IS NULL OR s.price <= :maxPrice)")
+    List<Service> searchServicesByPrice(@Param("categoryId") Long categoryId,
+                                           @Param("minPrice") Double minPrice,
+                                           @Param("maxPrice") Double maxPrice);
```

Comment:
Het complexe aan deze functionaliteit is dus dat de category, minPrice en maxPrice niet altijd zijn meegegeven. Kijk hoe je ze query dynamisch maakt.